mirascope/core/base/_extract.py

```diff
@@ -126,6 +129,12 @@ async def inner_async(
                 fields_from_call_args = get_fields_from_call_args(
                     response_model, fn, args, kwargs
                 )
+                if (
```

Comment:
is this necessary? even if the client is set in the decorator, if it's set in the dynamic config it'll get overwritten, so wouldn't `create_decorator` handle this for us already?